# HG changeset patch
# User Rik <rik@octave.org>
# Date 1422315169 28800
#      Mon Jan 26 15:32:49 2015 -0800
# Node ID 884e0c55d92c922d3d54220f5c3b527ab25d2b5e
# Parent  e68267373191cbe4b54b9b30c7ca0b19b8faa639
Fix complex compare operation for issorted (bug #44071).

* Array-C.cc (nan_ascending_compare, nan_descending_compare): Fix typo where
comparison was made between x and x rather than between x and y.

diff --git a/liboctave/array/Array-C.cc b/liboctave/array/Array-C.cc
--- a/liboctave/array/Array-C.cc
+++ b/liboctave/array/Array-C.cc
@@ -41,27 +41,27 @@ sort_isnan<Complex> (const Complex& x)
   return xisnan (x);
 }
 
 static bool
 nan_ascending_compare (const Complex& x, const Complex& y)
 {
   return (xisnan (y)
           ? ! xisnan (x)
-          : ((std::abs (x) < std::abs (x))
-             || ((std::abs (x) == std::abs (x)) && (arg (x) < arg (x)))));
+          : ((std::abs (x) < std::abs (y))
+             || ((std::abs (x) == std::abs (y)) && (arg (x) < arg (y)))));
 }
 
 static bool
 nan_descending_compare (const Complex& x, const Complex& y)
 {
   return (xisnan (x)
           ? ! xisnan (y)
-          : ((std::abs (x) > std::abs (x))
-             || ((std::abs (x) == std::abs (x)) && (arg (x) > arg (x)))));
+          : ((std::abs (x) > std::abs (y))
+             || ((std::abs (x) == std::abs (y)) && (arg (x) > arg (y)))));
 }
 
 Array<Complex>::compare_fcn_type
 safe_comparator (sortmode mode, const Array<Complex>& a , bool allow_chk)
 {
   Array<Complex>::compare_fcn_type result = 0;
 
   if (allow_chk)
